### ISSUE TITLE
Not installing extras / Behavior change from 1.0.10

### ISSUE THREAD

**Issue by Qu4tro** [python-poetry/poetry#3129]
- [x] I am on the [latest](https://github.com/python-poetry/poetry/releases/latest) Poetry version.
- [ ] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.

- **OS version and name**: Not relevant
- **Poetry version**: Latest
- **Link of a [Gist](https://gist.github.com/) with the contents of your pyproject.toml file**: Not relevant

## Issue
Hey. Heads up, it might be a duplicate, because there are multiple `extra` related issues and it's not trivial for me to parse them all. I'll however try to document my issue with a minimal and reproducible setup, so hopefully it will add some value here.

Let's start. 
### A story of two `poetry`
```
❯ ls
poetry1010  poetry1102

❯ ./poetry1010 -V && ./poetry1102 -V
Poetry version 1.0.10
Poetry version 1.1.2

❯ ./poetry1010 new help0 && ./poetry1102 new help1
Created package help0 in help0
Created package help1 in help1



❯ diff -u <(cd help0/ && ../poetry1010 add "dramatiq[rabbitmq]") <(cd help1/ && ../poetry1102 add "dramatiq[rabbitmq]" | tr -s "•" "-")
--- /proc/self/fd/12	2020-10-07 19:26:21.998079921 +0100
+++ /proc/self/fd/14	2020-10-07 19:26:21.998079921 +0100
@@ -1,18 +1,16 @@
-Creating virtualenv help0-oFs3ajBR-py3.8 in /home/quatro/.cache/pypoetry/virtualenvs
+Creating virtualenv help1-uRfcpVMu-py3.8 in /home/quatro/.cache/pypoetry/virtualenvs
 Using version ^1.9.0 for dramatiq
 
 Updating dependencies
 Resolving dependencies...
 
-
-Package operations: 12 installs, 0 updates, 0 removals
+Package operations: 11 installs, 0 updates, 0 removals
 
   - Installing pyparsing (2.4.7)
   - Installing six (1.15.0)
   - Installing attrs (20.2.0)
   - Installing more-itertools (8.5.0)
   - Installing packaging (20.4)
-  - Installing pika (1.1.0)
   - Installing pluggy (0.13.1)
   - Installing prometheus-client (0.8.0)
   - Installing py (1.9.0)
```

We can observe `pika` is no longer installed in the latest version.

- This later bit can be piped into `diff-so-fancy` for a better visual. 
- I use `rm -rf ~/.cache/pypoetry/virtualenvs/help*` to reset the virtualenvs.
- You can replace `poetry 1.1.2` with `poetry master` and it's the same outcome.

Let me know if there's anything else I can do to help or if I'm misunderstanding some new requirements .

**Comment by Qu4tro** [comment]
This might be a duplicate of https://github.com/python-poetry/poetry/issues/2992 .

I've managed to reproduce the previous behavior with:
```
docker run --rm -i --entrypoint bash python:3.8 <<EOF
set -e
python -m pip install -q poetry==1.1.0rc1
python -m poetry new foobar
pushd foobar
ls
sed -i /pytest/d pyproject.toml
python -m poetry add dramatiq[rabbitmq]
EOF
```

Replacing 1.1.0rc1 by 1.1.2 returns to the wrong behavior. 

Feel free to close the issue if I just need to wait for the next release.

**Comment by rickymyl-karhoo** [comment]
Not sure whether it is a duplicate hence posting here. I am trying to do

**Comment by abn** [comment]
@rickymyl-karhoo @Qu4tro can yo try the fix in #3147 please?

Note that #2992 is only partly related; the sisue there is alos the case of the extra package name used.

**Comment by rickymyl-karhoo** [comment]
How do you install a specific commit/branch? Thank you

**Comment by abn** [comment]
Something like this will work.

```console
podman run --rm -i --entrypoint bash python:3.8 <<EOF
set -e
python -m pip install -q git+https://github.com/python-poetry/poetry.git@refs/pull/3147/head
python -m poetry new foobar
pushd foobar
ls
sed -i /pytest/d pyproject.toml
python -m poetry add dramatiq[rabbitmq]
EOF
```

**Comment by Qu4tro** [comment]
Yep, I can confirm that it works with my setup. Thank you for the quick fix. 

I also want to say, that dependabot has updated to `Poetry 1.1` and has such, using it atm, might create an issue of missing extra packages (which was how I noticed this issue in the first place).

**Comment by abn** [comment]
Thank you for reporting this @Qu4tro. I might try get a release out soon. Hopefully not many projects get impacted by the dependabot update. :(

**Comment by ryanhiebert** [comment]
I'm trying to get switched to poetry today, and I ran into this issue. I'll try using an older version to see if that gets me unstuck for today.

**Comment by razorx89** [comment]
> Thank you for reporting this @Qu4tro. I might try get a release out soon. Hopefully not many projects get impacted by the dependabot update. :(

Any updates on this? I ran into this issue today, but unfortunately I cannot downgrade to 1.0.10 due to another bug.

**Comment by abn** [comment]
@razorx89 I am waiting on reviews for the fixes; unfortunately the team is a small group of folks volunteering spare time. :)

**Comment by mr-bjerre** [comment]
How come this was closed? I'm still having this issue in 1.1.3

**Comment by ryanhiebert** [comment]
1.1.3 fixed the issue in my case. There might be something different about your setup that needs to be identified.
